add demex

eugenelhj opened the pull request:
hi, adding yield for demex app , under carbon chain (cosmos)

llamatester commented:
The demex adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       354 passed, 354 total
    Snapshots:   0 total
    Time:        0.462 s
    Ran all test suites.
    
    Nb of pools: 50
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────┬──────────┬─────────┬────────────────────────────────────────────────────────────────────────────────────────────┬──────────────────────────────────────────────┬────────────────────┬────────────────────┬──────────────┬────────────────────────────────────────────────────────────────────────────────────┐
    │ (index) │                     pool                      │  chain   │ project │                                          poolMeta                                          │                    symbol                    │       tvlUsd       │        apy         │ rewardTokens │                                  underlyingTokens                                  │
    ├─────────┼───────────────────────────────────────────────┼──────────┼─────────┼────────────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────────────────┼────────────────────┼────────────────────┼──────────────┼────────────────────────────────────────────────────────────────────────────────────┤
    │    0    │  'wbtc.1.2.786598-50-eth.1.2.942d87-50-lp3'   │ 'Carbon' │ 'demex' │                      '50% WBTC.1.2.786598 / 50% ETH.1.2.942D87 Pool'                       │         '% WBTC... / % ETH...D Pool'         │ 236996.3231476601  │ 4.965449137085113  │  [ 'swth' ]  │                      [ 'wbtc.1.2.786598', 'eth.1.2.942d87' ]                       │
    │    1    │       'busd.1.6.754a80-50-swth-50-lp15'       │ 'Carbon' │ 'demex' │                           '50% BUSD.1.6.754A80 / 50% SWTH Pool'                            │          '% BUSD...A / % SWTH Pool'          │ 214498.47507012662 │ 43.000764495649676 │  [ 'swth' ]  │                           [ 'busd.1.6.754a80', 'swth' ]                            │
    │    2    │       'usdc.1.2.343151-50-swth-50-lp30'       │ 'Carbon' │ 'demex' │                           '50% USDC.1.2.343151 / 50% SWTH Pool'                            │          '% USDC... / % SWTH Pool'           │ 203852.36154595335 │ 44.81685282612468  │  [ 'swth' ]  │                           [ 'usdc.1.2.343151', 'swth' ]                            │
    │    3    │      'zwap.1.18.393529-50-swth-50-lp41'       │ 'Carbon' │ 'demex' │                           '50% ZWAP.1.18.393529 / 50% SWTH Pool'                           │          '% ZWAP... / % SWTH Pool'           │ 144127.61207629406 │ 31.939643540855158 │  [ 'swth' ]  │                           [ 'zwap.1.18.393529', 'swth' ]                           │
    │    4    │                   'clpt/44'                   │ 'Carbon' │ 'demex' │ '50% IBC/D70B0FBF97AEB04491E9ABF4467A7F66CD6250F4382CE5192D856114B83738D2 / 50% SWTH Pool' │ '% IBC/DBFBFAEBEABFAFCDFCEDBD / % SWTH Pool' │ 127932.39729407086 │         0          │  [ 'swth' ]  │ [ 'ibc/D70B0FBF97AEB04491E9ABF4467A7F66CD6250F4382CE5192D856114B83738D2', 'swth' ] │
    │    5    │  'usdc.1.2.343151-50-wbtc.1.2.786598-50-lp5'  │ 'Carbon' │ 'demex' │                      '50% USDC.1.2.343151 / 50% WBTC.1.2.786598 Pool'                      │         '% USDC... / % WBTC... Pool'         │  127106.684647428  │ 18.219640088620793 │  [ 'swth' ]  │                      [ 'usdc.1.2.343151', 'wbtc.1.2.786598' ]                      │
    │    6    │       'ava1.1.6.95712b-50-swth-50-lp40'       │ 'Carbon' │ 'demex' │                           '50% AVA1.1.6.95712B / 50% SWTH Pool'                            │          '% AVA...B / % SWTH Pool'           │ 126946.60915560869 │ 38.39264275959823  │  [ 'swth' ]  │                           [ 'ava1.1.6.95712b', 'swth' ]                            │
    │    7    │  'busd.1.6.754a80-50-bnb.1.6.773edb-50-lp22'  │ 'Carbon' │ 'demex' │                      '50% BUSD.1.6.754A80 / 50% BNB.1.6.773EDB Pool'                       │       '% BUSD...A / % BNB...EDB Pool'        │ 90604.64350037713  │ 25.597606238216546 │  [ 'swth' ]  │                      [ 'busd.1.6.754a80', 'bnb.1.6.773edb' ]                       │
    │    8    │  'usdc.1.2.343151-50-eth.1.2.942d87-50-lp4'   │ 'Carbon' │ 'demex' │                      '50% USDC.1.2.343151 / 50% ETH.1.2.942D87 Pool'                       │         '% USDC... / % ETH...D Pool'         │ 84013.66286616698  │ 27.940932125182176 │  [ 'swth' ]  │                      [ 'usdc.1.2.343151', 'eth.1.2.942d87' ]                       │
    │    9    │ 'xsgd.1.18.be52cd-50-busd.1.6.754a80-50-lp33' │ 'Carbon' │ 'demex' │                     '50% XSGD.1.18.BE52CD / 50% BUSD.1.6.754A80 Pool'                      │      '% XSGD...BECD / % BUSD...A Pool'       │ 67389.04798796981  │         0          │  [ 'swth' ]  │                     [ 'xsgd.1.18.be52cd', 'busd.1.6.754a80' ]                      │
    └─────────┴───────────────────────────────────────────────┴──────────┴─────────┴────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────┴────────────────────┴────────────────────┴──────────────┴────────────────────────────────────────────────────────────────────────────────────┘

vrtnd commented:
Hi, thanks for the PR.
Please format `symbol` field   `"% WBTC... / % ETH...D Pool"  ->  "WBTC-ETH"`  and remove meta field, because it is just duplicate of `symbol` feld

eugenelhj commented:
> Hi, thanks for the PR. Please format `symbol` field `"% WBTC... / % ETH...D Pool" -> "WBTC-ETH"` and remove meta field, because it is just duplicate of `symbol` field
> 
> Is `swth` a correct token addres for carbon blockhain?

Hi, thanks for the prompt response. Fixed the symbol and removed meta field. afaik , cosmos chain does not have token address and used denom directly eg. https://api.carbon.network/cosmos/bank/v1beta1/supply/swth

llamatester commented:
The demex adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       354 passed, 354 total
    Snapshots:   0 total
    Time:        0.522 s
    Ran all test suites.
    
    Nb of pools: 50
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────┬──────────┬─────────┬─────────────┬────────────────────┬────────────────────┬──────────────┬────────────────────────────────────────────────────────────────────────────────────┐
    │ (index) │                     pool                      │  chain   │ project │   symbol    │       tvlUsd       │        apy         │ rewardTokens │                                  underlyingTokens                                  │
    ├─────────┼───────────────────────────────────────────────┼──────────┼─────────┼─────────────┼────────────────────┼────────────────────┼──────────────┼────────────────────────────────────────────────────────────────────────────────────┤
    │    0    │  'wbtc.1.2.786598-50-eth.1.2.942d87-50-lp3'   │ 'Carbon' │ 'demex' │ 'WBTC-ETH'  │ 236996.0860399401  │ 4.903472812595279  │  [ 'swth' ]  │                      [ 'wbtc.1.2.786598', 'eth.1.2.942d87' ]                       │
    │    1    │       'busd.1.6.754a80-50-swth-50-lp15'       │ 'Carbon' │ 'demex' │ 'BUSD-SWTH' │ 214497.98862393925 │ 42.91353500203095  │  [ 'swth' ]  │                           [ 'busd.1.6.754a80', 'swth' ]                            │
    │    2    │       'usdc.1.2.343151-50-swth-50-lp30'       │ 'Carbon' │ 'demex' │ 'USDC-SWTH' │ 203974.94129868428 │ 44.68121132485256  │  [ 'swth' ]  │                           [ 'usdc.1.2.343151', 'swth' ]                            │
    │    3    │      'zwap.1.18.393529-50-swth-50-lp41'       │ 'Carbon' │ 'demex' │ 'ZWAP-SWTH' │ 144914.46580123383 │ 31.853854698621568 │  [ 'swth' ]  │                           [ 'zwap.1.18.393529', 'swth' ]                           │
    │    4    │                   'clpt/44'                   │ 'Carbon' │ 'demex' │ 'LUNA-SWTH' │ 127932.39729407086 │         0          │  [ 'swth' ]  │ [ 'ibc/D70B0FBF97AEB04491E9ABF4467A7F66CD6250F4382CE5192D856114B83738D2', 'swth' ] │
    │    5    │  'usdc.1.2.343151-50-wbtc.1.2.786598-50-lp5'  │ 'Carbon' │ 'demex' │ 'USDC-WBTC' │  127107.531938722  │ 18.070867451487448 │  [ 'swth' ]  │                      [ 'usdc.1.2.343151', 'wbtc.1.2.786598' ]                      │
    │    6    │       'ava1.1.6.95712b-50-swth-50-lp40'       │ 'Carbon' │ 'demex' │ 'AVA1-SWTH' │ 126946.60915560869 │ 38.12834311807133  │  [ 'swth' ]  │                           [ 'ava1.1.6.95712b', 'swth' ]                            │
    │    7    │  'busd.1.6.754a80-50-bnb.1.6.773edb-50-lp22'  │ 'Carbon' │ 'demex' │ 'BUSD-BNB'  │ 90604.64350037713  │ 25.344671240080192 │  [ 'swth' ]  │                      [ 'busd.1.6.754a80', 'bnb.1.6.773edb' ]                       │
    │    8    │  'usdc.1.2.343151-50-eth.1.2.942d87-50-lp4'   │ 'Carbon' │ 'demex' │ 'USDC-ETH'  │ 84011.33910943098  │  27.6519465017561  │  [ 'swth' ]  │                      [ 'usdc.1.2.343151', 'eth.1.2.942d87' ]                       │
    │    9    │ 'xsgd.1.18.be52cd-50-busd.1.6.754a80-50-lp33' │ 'Carbon' │ 'demex' │ 'XSGD-BUSD' │ 67389.04798796981  │         0          │  [ 'swth' ]  │                     [ 'xsgd.1.18.be52cd', 'busd.1.6.754a80' ]                      │
    └─────────┴───────────────────────────────────────────────┴──────────┴─────────┴─────────────┴────────────────────┴────────────────────┴──────────────┴────────────────────────────────────────────────────────────────────────────────────┘

eugenelhj commented:
Hello @vrtnd  , can you kindly check why does only 19 pools appear under demex? We have over 50 now and the above test shows atleast 50 as well

vrtnd commented:
We have minimum tvl filter >$10k to display on UI
